File: go/diagnostics.go
package jrmclient

import (
	"context"
	"encoding/json"
	"fmt"
)

// Ping performs a trivial JWT-authenticated liveness check via
// GET /api/v1/ping, returning the caller's identity and tier as decoded
// from the bearer token.
func (c *Client) Ping(ctx context.Context) (*PingResponse, error) {
	var out PingResponse
	if err := c.doRequest(ctx, "GET", "/api/v1/ping", nil, nil, requestOptions{
		authorize: true,
	}, &out); err != nil {
		return nil, err
	}
	return &out, nil
}

// Echo sends an arbitrary JSON-serializable value to POST /api/v1/echo and
// returns it verbatim as decoded by the server, useful for confirming
// request encoding and auth header handling end to end. Requires a bearer
// token.
func (c *Client) Echo(ctx context.Context, payload any) (*EchoResponse, error) {
	body, err := json.Marshal(payload)
	if err != nil {
		return nil, fmt.Errorf("jrmclient: encoding request: %w", err)
	}

	var out EchoResponse
	if err := c.doRequest(ctx, "POST", "/api/v1/echo", nil, body, requestOptions{
		contentType: "application/json",
		authorize:   true,
	}, &out); err != nil {
		return nil, err
	}
	return &out, nil
}

File: python/src/jrm_code_client/diagnostics.py
"""GET /api/v1/ping and POST /api/v1/echo"""

from __future__ import annotations

from typing import Any

from .client import _BaseClient
from .models import EchoResponse, PingResponse


class _DiagnosticsMixin(_BaseClient):
    def ping(self) -> PingResponse:
        """Perform a trivial JWT-authenticated liveness check via
        GET /api/v1/ping, returning the caller's identity and tier as
        decoded from the bearer token. Requires ``client.token`` to be
        set."""
        data = self._request_json("GET", "/api/v1/ping", authorize=True)
        return PingResponse._from_json(data)

    def echo(self, payload: Any) -> EchoResponse:
        """Send PAYLOAD (any JSON-serializable value) to POST /api/v1/echo
        and return it verbatim as decoded by the server, useful for
        confirming request encoding and auth header handling end to end.
        Requires ``client.token`` to be set."""
        data = self._request_json(
            "POST",
            "/api/v1/echo",
            json_body=payload,
            content_type="application/json",
            authorize=True,
        )
        return EchoResponse._from_json(data)
